src/main/java/org/janelia/saalfeldlab/n5/util/Copy.java

```diff
@@ -0,0 +1,496 @@
+	 *            N5 root to copy from
+	 * @param target
+	 *            N5 root to copy into
+	 * @param dataset
+	 *            Data set to be copied
+	 * @param attributes
+	 *            Data set attributes
+	 * @param startBlock
+	 * @param stopBlock
+	 * @throws DatasetDoesNotExistException
+	 *             Data set does not exist in {@code source}
+	 * @throws IOException
+	 */
+	public static void copyWithoutCreation(
+			final N5Reader source,
+			final N5Writer target,
+			final String dataset,
+			final DatasetAttributes attributes,
+			final long startBlock,
+			final long stopBlock ) throws DatasetDoesNotExistException, IOException
+	{
+		final long[] gridDims = attributes.getDimensions().clone();
+		divideInPlace( gridDims, attributes.getBlockSize() );
+		final long[] blockPosition = new long[ gridDims.length ];
+		for ( long block = startBlock; block < stopBlock; ++block )
+		{
+			indexToPosition( block, gridDims, blockPosition );
+			final DataBlock< ? > data = source.readBlock( dataset, attributes, blockPosition );
+			if ( data != null )
+			{
+				target.writeBlock( dataset, attributes, data );
+			}
+		}
+	}
+
+	/**
+	 * Copy all attributes ignoring data set attributes if group is a data set.
+	 *
+	 * @param source
+	 * @param target
+	 * @param groupName
+	 * @throws IOException
+	 */
+	public static void copyAttributes( final N5Reader source, final N5Writer target, final String groupName )
+			throws IOException
+	{
+
+		final Map< String, Class< ? > > attributes = source.listAttributes( groupName );
+
+		final DatasetAttributes datasetAttributes = source.datasetExists( groupName ) ? source.getDatasetAttributes( groupName ) : null;
+		final Set< String > datasetAttributeKeys = datasetAttributes == null ? new HashSet<>() : datasetAttributes.asMap().keySet();
+		attributes
+		.entrySet()
+		.stream()
+		.filter( e -> !datasetAttributeKeys.contains( e.getKey() ) )
+		.forEach( unchecked( e -> target.setAttribute( groupName, e.getKey(), source.getAttribute( groupName, e.getKey(), e.getValue() ) ) ) );
+	}
+
+	/**
+	 * list all contained groups/data sets
+	 *
+	 * @param n5
+	 * @return
+	 * @throws IOException
+	 */
+	public static List< String > listAll( final N5Reader n5 ) throws IOException
+	{
+		return listAll( n5, "" );
+	}
+
+	/**
+	 * list all contained groups/data sets starting at {@code from}
+	 *
+	 * @param n5
+	 * @param from
+	 * @return
+	 * @throws IOException
+	 */
+	public static List< String > listAll( final N5Reader n5, final String from ) throws IOException
+	{
+		final List< String > store = new ArrayList<>();
+		listAll( n5, from, store );
+		return store;
+	}
+
+	private static void listAll( final N5Reader n5, final String from, final List< String > store ) throws IOException
+	{
+		store.add( from );
+		if ( n5.datasetExists( from ) ) { return; }
+
+		final String[] subGroups = n5.list( from );
+		for ( final String g : subGroups )
+		{
+			listAll( n5, Paths.get( from, g ).toString(), store );
```

Comment:
There was an issue on Windows that `Paths.get(from, ...)` calls fail with `InvalidPathException` when `from` starts with a slash character.
N5 seems to generally allow both uses (e.g. `group/dataset` and `/group/dataset`), so I addressed this in #30 by effectively removing the leading slash if it's there to ensure that concatenation works properly. In case someone is going to use it on Windows, it may be helpful to integrate the same fix here.

Reply:
That's a good catch, I was unaware of this. I will make sure to remove leading slashes.